src/vs/workbench/contrib/welcome/banner/browser/welcomeBanner.contribution.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { LifecyclePhase } from 'vs/workbench/services/lifecycle/common/lifecycle';
+import { Registry } from 'vs/platform/registry/common/platform';
+import { Extensions as WorkbenchExtensions, IWorkbenchContributionsRegistry } from 'vs/workbench/common/contributions';
+import { IBannerService } from 'vs/workbench/services/banner/browser/bannerService';
+import { Codicon } from 'vs/base/common/codicons';
+import { IStorageService, StorageScope, StorageTarget } from 'vs/platform/storage/common/storage';
+import { IWorkbenchEnvironmentService } from 'vs/workbench/services/environment/common/environmentService';
+
+class WelcomeBannerContribution {
+
+	private static readonly WELCOME_BANNER_DISMISSED_KEY = 'workbench.banner.welcome.dismissed';
+
+	constructor(
+		@IBannerService bannerService: IBannerService,
+		@IStorageService storageService: IStorageService,
+		@IWorkbenchEnvironmentService environmentService: IWorkbenchEnvironmentService
+	) {
+		const welcomeBanner = environmentService.options?.welcomeBanner;
+		if (!welcomeBanner) {
+			return; // welcome banner is not enabled
+		}
+
+		if (storageService.getBoolean(WelcomeBannerContribution.WELCOME_BANNER_DISMISSED_KEY, StorageScope.GLOBAL, false)) {
+			return; // welcome banner dismissed
+		}
+
+		bannerService.show({
+			id: 'welcome.banner',
+			message: welcomeBanner.message,
+			icon: Codicon.code,
```

Comment:
Should we make the icon customizable as well? Most consumers might be happy with the logo, so this might be also a follow up as the need arises.

Reply:
Added that :thu